### PR TITLE
Add README file for lightning image segmentation workload 

### PR DESCRIPTION
**Pull request by abhibyreddi**

### CODE REVIEW

File: demo/lightning/image-segmentation/README.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
+# Image Segmentation Demo Code
+The code examples in this directory demonstrate how Dataflux can be used for image segmentation training alongside PyTorch Lightning. The image segmentation workload implemented here works on the KiTS 19 dataset which has `210` images and their corresponding labels. The images and their labels are stored in separate directories.
```
Comment (jdnurme) [inline demo/lightning/image-segmentation/README.md:2]: i would say "which contains 210 images" for clarity.

File: demo/lightning/image-segmentation/README.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+                    self.train_dataset,
+                    seed=self.args.seed,
+                    drop_last=True
+                )
+    
+    ```
+
+1. `train.py`
+
+    An instance of the `Unet3DDataModule` class is created and passed to the Pytorch Lightning trainer instance's `fit` function as the `train_dataloaders`
```
Comment (jdnurme) [inline demo/lightning/image-segmentation/README.md:78]: "in the train_dataloader keyword argument"

File: demo/lightning/image-segmentation/README.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    )
+    trainer.fit(model=model, train_dataloaders=train_data_loader)
+
+    ```
+
+
+Please note that these demos are for educational and example purposes only, and have not been optimized for performance.
+
+
+## Setup
```
Comment (jdnurme) [inline demo/lightning/image-segmentation/README.md:97]: This section is great, but since users may or may not be on unix based systems i'm not sure if we should include the specific commands. @MattIrv ?

File: demo/lightning/image-segmentation/README.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    --prefetch_factor=5 \
+    --num_devices=1 \
+    --num_nodes=1 \
+    --local=True 
+```
+
+Be sure to set `--local` to `True`.
+
+## Multi-node GKE Cluster Execution
+_Note: the following instructions assume that you have Jobset and Kueue enabled on your GKE cluster. For easy compatability we recommend creating a cluster with XPK which will configure these features automatically._
```
Comment (jdnurme) [inline demo/lightning/image-segmentation/README.md:137]: maybe hyperlink XPK here?

File: demo/lightning/image-segmentation/README.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+    # If needed, run
+    gcloud auth login && gcloud auth application-default login
+    # Connect to the cluster
+    gcloud container clusters get-credentials {YOUR-GKE-CLUSTER-NAME} --zone {ZONE} --project {YOUR-GCP-PROJECT}
+    ```
+
+1. Build the demo container
+
+    Make sure your working directory is `dataflux-pytorch`
+    ```sh
```
Comment (jdnurme) [inline demo/lightning/image-segmentation/README.md:150]: Same question as above since users may not be on unix.

**Review by jdnurme — APPROVED** [review]

File: demo/lightning/image-segmentation/README.md (hunk 2 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done.

File: demo/lightning/image-segmentation/README.md (hunk 1 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done.

File: demo/lightning/image-segmentation/README.md (hunk 4 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done.

File: demo/lightning/image-segmentation/README.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
+# Image Segmentation Demo Code
+The code examples in this directory demonstrate how Dataflux can be used for image segmentation training alongside PyTorch Lightning. The image segmentation workload implemented here works on the [KiTS19](https://github.com/neheller/kits19) dataset which contains `210` images and their corresponding labels. The images and their labels are stored in separate directories.
```
Comment (bernardhan33) [inline demo/lightning/image-segmentation/README.md:2]: Do we still want the "Dataflux" naming?

File: demo/lightning/image-segmentation/README.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+    ```
+
+
+Please note that these demos are for educational and example purposes only, and have not been optimized for performance.
+
+
+## Setup
+1. Create a copy of the dataset in your GCP project
+    ```sh
+    gcloud storage cp -r gs://dataflux-demo-public/image-seg-dataset gs://{YOUR-GCS-BUCKET}
```
Comment (bernardhan33) [inline demo/lightning/image-segmentation/README.md:100]: Did I miss it? I'm not seeing the directory [here](https://pantheon.corp.google.com/storage/browser/dataflux-demo-public;tab=objects?forceOnBucketsSortingFiltering=true&e=-13802955&mods=monitoring_api_prod&project=dataflux-project&prefix=&forceOnObjectsSortingFiltering=false&pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))). The command will also fail.

File: demo/lightning/image-segmentation/README.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+    ```sh
+    # Create a new directory on your workstation
+    mkdir dataflux-demo && cd dataflux-demo
+
+    # Clone dataflux-pytorch
+    git clone --recurse-submodules https://github.com/GoogleCloudPlatform/dataflux-pytorch
+    ```
+1. Install dependencies
+    ```sh
+    # Create a python virutal environment and activate it
```
Comment (bernardhan33) [inline demo/lightning/image-segmentation/README.md:112]: nit: typo on `virtual`.

**Review by bernardhan33 — APPROVED** [review]

File: demo/lightning/image-segmentation/README.md (hunk 7 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md:100]: I haven't created it yet. Couldn't find the bucket in our google.com project. Was going to reach out to you tomorrow :) I will copy the dataset over into it now. Thanks!

File: demo/lightning/image-segmentation/README.md (hunk 6 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md:2]: Should I change it to "...demonstrate how GCS Connector for Pytorch (previously known as Dataflux)..." or just "...GCS Connector for Pytorch" ?

File: demo/lightning/image-segmentation/README.md (hunk 8 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done!

File: demo/lightning/image-segmentation/README.md (hunk 7 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done!

File: demo/lightning/image-segmentation/README.md (hunk 6 above)
Comment (MattIrv) [reply demo/lightning/image-segmentation/README.md:2]: The latter sounds good to me

File: demo/lightning/image-segmentation/README.md (hunk 3 above)
Comment (MattIrv) [reply demo/lightning/image-segmentation/README.md:97]: I think it's fine for a demo, any Windows user should be able to easily translate them

File: demo/lightning/image-segmentation/README.md (hunk 6 above)
Comment (abhibyreddi) [reply demo/lightning/image-segmentation/README.md]: Done
